core/src/main/java/org/apache/brooklyn/core/mgmt/internal/BrooklynShutdownHooks.java

```diff
@@ -100,6 +99,17 @@ public static void invokeStopOnShutdown(Entity entity) {
         }
     }
 
+    public static void resetShutdownFlag() {
```

Comment:
might be nice to make this `resetShutdownFlag(boolean newValue)` and then also use it in [run](https://github.com/m4rkmckenna/brooklyn-server/blob/7068308917b7424a4ff3cdab1f4416e45508c399/core/src/main/java/org/apache/brooklyn/core/mgmt/internal/BrooklynShutdownHooks.java#L190-L197).